src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx

```diff
@@ -182,13 +186,16 @@ function ProviderSelectionField({
                       <button
                         type="button"
                         onClick={() => {
-                          onChange(
-                            communityCareProviderList.find(
-                              p => p.id === checkedProvider,
-                            ),
+                          const providerIndex = communityCareProviderList.findIndex(
```

Comment:
In the `currentlyShownProvidersList.map(provider => {` call above, there's a [second argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) that's the current index. So  `currentlyShownProvidersList.map((provider, providerIndex) => {` would provide you with the index you need for the event.

Reply:
Good call, I didn't think about that